bittensor/subtensor.py

```diff
@@ -1319,6 +1319,71 @@ def root_set_weights(
             wait_for_finalization=wait_for_finalization,
             prompt=prompt,
         )
+    
+    ########################
+    #### Registry Calls ####
+    ########################
+
+    """ Queries subtensor registry named storage with params and block. """
+
+    def query_identity(
+        self,
+        key: str,
+        block: Optional[int] = None,
+    ) -> Optional[object]:
+        @retry(delay=2, tries=3, backoff=2, max_delay=4)
+        def make_substrate_call_with_retry():
+            with self.substrate as substrate:
+                return substrate.query(
+                    module="Registry",
+                    storage_function='IdentityOf',
+                    params=[key],
+                    block_hash=None
+                    if block == None
+                    else substrate.get_block_hash(block),
+                )
+            
+        identity_info  = make_substrate_call_with_retry()
+        return bittensor.utils.wallet_utils.decode_hex_identity_dict(identity_info.value['info'])  
+
+    def update_identity(
+        self,
+        wallet: "bittensor.wallet",
```

Comment:
If the wallet coldkey is not the subnet owner, this can essentially set identity for any coldkey on chain. Do we want to limit this to subnet owners only? (check the chain for current subnet coldkey owners)

E.g.:
```python
owner_coldkeys = [sub.get_subnet_owner(i) for i in sub.get_all_subnet_netuids()]
if wallet.coldkey.ss58_address not in owner_coldkeys:
   raise ...
```

Reply:
Yeap, this is done by the chain, only subnet owners and root validators can set an identity